In [24]:
import numpy as np
import pandas as pd
import statsmodels.api as sm

Extraction des Outputs de 2022, dans un premiers temps on s'intéresse aux services d'Urgence, de MCO et de SSR. On pourra ajouter d'autres services par la suite (Psychiatrie par exemple)

In [25]:
FINESS = pd.read_excel("finess.xlsx")
FINESS = FINESS.drop(FINESS.columns[[1,2,4]],axis=1)
FINESS = FINESS.rename(columns={"FINESS":"FI","Statut Juridique":"Statut"})

SYGEN2022 = pd.read_csv("SAE/2022/SYGEN_2022r.csv", sep=";", encoding="latin-1")
Data = SYGEN2022[['FI','ETPSAL_TOT','EFFLIB_TOT','ETP_INFSANSSPE','ETP_INFAVECSPE','ETP_AID','ETP_DIR','ETP_DIRSOI','ETP_AUTADM','SEJHC_SSR','SEJHC_MCO','SEJHP_MCO','SEJ_HTP_TOT','VEN_HDJ_TOT','VEN_HDN_TOT','SEAN_HEMO_CENTRE','SEAN_CHIMIO','SEAN_RADIO']]
Data = Data.merge(FINESS, on="FI", how="left")

In [26]:
Data.describe()

,ETPSAL_TOT,EFFLIB_TOT,ETP_INFSANSSPE,ETP_INFAVECSPE,ETP_AID,ETP_DIR,ETP_DIRSOI,ETP_AUTADM,SEJHC_SSR,SEJHC_MCO,SEJHP_MCO,SEJ_HTP_TOT,VEN_HDJ_TOT,VEN_HDN_TOT,SEAN_HEMO_CENTRE,SEAN_CHIMIO,SEAN_RADIO
count,3489.000000,1473.000000,3639.00000,1204.000000,3167.000000,2789.000000,1151.000000,3549.000000,1802.000000,1636.00000,1636.000000,512.000000,529.000000,210.000000,337.000000,728.000000,204.000000
mean,26.105374,28.909029,79.71737,25.843164,73.327335,2.790470,0.918566,37.527954,474.011654,6219.24022,4988.977995,1027.726562,7061.068053,399.300000,11950.632047,4425.013736,20058.475490
std,70.215511,46.663616,167.04529,48.049243,141.277774,3.788046,0.655406,80.022563,379.674641,9561.88067,6781.532895,1060.630557,6594.686080,1276.662635,7197.093938,5731.908423,13783.578855
min,0.000000,0.000000,0.01000,0.000000,0.050000,0.010000,0.000000,0.010000,0.000000,0.00000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,1.330000,3.000000,8.80000,1.927500,11.740000,1.000000,0.500000,3.300000,220.000000,321.00000,0.000000,395.750000,2251.000000,0.000000,7497.000000,450.500000,10787.250000
50%,3.890000,7.000000,20.00000,7.595000,26.500000,1.750000,1.000000,9.150000,377.000000,2183.00000,2400.000000,724.000000,5167.000000,0.000000,10988.000000,2281.000000,19352.500000
75%,10.980000,35.000000,59.53500,27.162500,65.925000,3.460000,1.000000,29.960000,633.750000,8175.25000,7426.000000,1320.750000,10084.000000,156.500000,15249.000000,6169.500000,28604.000000
max,949.030000,321.000000,1912.95000,542.870000,1863.290000,61.580000,6.000000,868.120000,2755.000000,75761.00000,54968.000000,11582.000000,41548.000000,12320.000000,47586.000000,37982.000000,61278.000000


In [27]:
Urg2022 = pd.read_csv("SAE/2022/URGENCES2_2022r.csv", sep=";", encoding="latin-1")
Urg2022 = Urg2022[['FI','PASSU']]
Data = Data.merge(Urg2022, on="FI", how="left")

In [28]:
USLD2022 = pd.read_csv("SAE/2022/USLD_2022r.csv", sep=";", encoding="latin-1")
USLD2022 = USLD2022[['FI','ENT']]
Data = Data.merge(USLD2022, on="FI", how="left")

In [29]:
SYGEN2022

,FI,FI_EJ,AN,LIT_MED,LIT_CHI,LIT_OBS,LIT_MCO,SEJHC_MED,SEJHC_CHI,SEJHC_OBS,...,ACT_SCAN,NB_IRM,ACT_IRM,NB_CAM,ACT_CAM,NB_TOMO,ACT_TOMO,SAL_INTERV,POST_REVEIL,BOR
0,010000024,010780054,2022,270.0,68.0,47.0,385.0,14972.0,7260.0,3474.0,...,1.0,3.0,0.0,NaN,NaN,NaN,NaN,13.0,19.0,SYGEN
1,010000032,010780062,2022,66.0,17.0,8.0,91.0,4020.0,970.0,501.0,...,1.0,NaN,NaN,NaN,NaN,NaN,NaN,4.0,6.0,SYGEN
2,010000065,010780096,2022,59.0,NaN,NaN,59.0,1669.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN
3,010000081,010780112,2022,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN
4,010000099,010780120,2022,10.0,NaN,NaN,10.0,170.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3983,980500763,970407250,2022,NaN,NaN,NaN,NaN,0.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN
3984,980500920,970407250,2022,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN
3985,980501159,970407250,2022,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN
3986,980501258,970407250,2022,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,SYGEN


Total médecins (libéraux et saliariés confondus) : 

In [30]:
Data['MED'] = Data['ETPSAL_TOT'] + Data['EFFLIB_TOT']

Total infirmiers (spécialisés et non-spécialisés) : 

In [31]:
Data['IDE'] = Data['ETP_INFSANSSPE'] + Data['ETP_INFAVECSPE']

Total personnel administratif :

In [32]:
Data['ADMIN'] = Data['ETP_DIR'] + Data['ETP_DIRSOI'] + Data['ETP_AUTADM']

Total séjours MCO (hospitalisation partielle et complète confondu) :

In [33]:
Data['MCO'] = Data['SEJHC_MCO'] + Data['SEJHP_MCO']

Total séjours en psychiatrie (hospitalisations complètes, venues de jour et venues de nuit) :

In [34]:
Data['PSY'] = Data['SEJ_HTP_TOT'] + Data['VEN_HDJ_TOT'] + Data['VEN_HDN_TOT']

In [35]:
Data["SEANCES"] = Data['SEAN_HEMO_CENTRE'] + Data['SEAN_CHIMIO'] + Data['SEAN_RADIO']

Passage des variables d'intérêt en log (convention log(0) = 0) :

In [36]:
Data["lURG"] = np.where(Data["PASSU"] > 0, np.log(Data["PASSU"]), 0)
Data["lMCO"] = np.where(Data["MCO"] > 0, np.log(Data["MCO"]), 0)
Data["lPSY"] = np.where(Data["PSY"] > 0, np.log(Data["PSY"]), 0)
Data["lSSR"] = np.where(Data["SEJHC_SSR"] > 0, np.log(Data["SEJHC_SSR"]), 0)
Data["lSEANCES"] = np.where(Data["SEANCES"] > 0, np.log(Data["SEANCES"]), 0)
Data['lUSLD'] = np.where(Data['ENT'] > 0, np.log(Data['ENT']), 0)

Data["lMED"] = np.where(Data["MED"] > 0, np.log(Data["MED"]), 0)
Data["lIDE"] = np.where(Data["IDE"] > 0, np.log(Data["IDE"]), 0)
Data["lAID"] = np.where(Data["ETP_AID"] > 0, np.log(Data["ETP_AID"]), 0)
Data["lADMIN"] = np.where(Data["ADMIN"] > 0, np.log(Data["ADMIN"]), 0)

c:\Users\Alexandre\anaconda3\Lib\site-packages\pandas\core\arraylike.py:399: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)
c:\Users\Alexandre\anaconda3\Lib\site-packages\pandas\core\arraylike.py:399: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)
c:\Users\Alexandre\anaconda3\Lib\site-packages\pandas\core\arraylike.py:399: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)
c:\Users\Alexandre\anaconda3\Lib\site-packages\pandas\core\arraylike.py:399: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)
c:\Users\Alexandre\anaconda3\Lib\site-packages\pandas\core\arraylike.py:399: RuntimeWarning: divide by zero encountered in log
  result = getattr(ufunc, method)(*inputs, **kwargs)


On construit deux variables de contrôle : Statut_PNL (resp. Statut_PL) vaut 1 si l'établissement est privé non lucratif (resp. privé lucratif) 

In [37]:
dummies = pd.get_dummies(Data["Statut"], prefix="Statut")

Data["Statut_PNL"] = dummies["Statut_Privé non lucratif"]
Data["Statut_PL"]  = dummies["Statut_Privé lucratif"]

Data["Statut_PNL"] = Data["Statut_PNL"].astype(int)
Data["Statut_PL"]  = Data["Statut_PL"].astype(int)



Catégorisation des variables :

In [39]:
outputs = ["lURG", "lMCO", "lSSR", "lPSY","lSEANCES", "lUSLD"]
controls = ["Statut_PNL", "Statut_PL"]
inputs = ["lMED", "lIDE", "lAID", "lADMIN"]

Estimation des coefficients :

In [40]:
results = {}
dict = {}

for var in outputs:
    Data[f"{var}_PNL"] = Data[var] * Data["Statut_PNL"]
    Data[f"{var}_PL"]  = Data[var] * Data["Statut_PL"]


for inp in inputs:
    X = Data[outputs + controls + [f"{v}_PNL" for v in outputs] + [f"{v}_PL" for v in outputs]]
    X = sm.add_constant(X)
    y = Data[inp]
    
    model = sm.OLS(y, X).fit(cov_type="HC1")  # erreurs robustes
    results[inp] = model
    
    print("\n" + "="*60)
    print(f"Demande conditionnelle pour : {inp}")
    print(model.summary())
    
    print(inp)
    for c in controls:
        coef = model.params[c]
        pct = (np.exp(coef) - 1) * 100
        print(f"{c} → différence d'utilisation : {pct:.2f}%")
        dict[(inp,c)] = pct


Demande conditionnelle pour : lMED
                            OLS Regression Results                            
Dep. Variable:                   lMED   R-squared:                       0.472
Model:                            OLS   Adj. R-squared:                  0.470
Method:                 Least Squares   F-statistic:                     7671.
Date:                Wed, 04 Feb 2026   Prob (F-statistic):               0.00
Time:                        14:21:11   Log-Likelihood:                -6610.5
No. Observations:                4058   AIC:                         1.326e+04
Df Residuals:                    4037   BIC:                         1.340e+04
Df Model:                          20                                         
Covariance Type:                  HC1                                         
                   coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------
const       

In [41]:
dict

{('lMED', 'Statut_PNL'): 9.046198853971998,
 ('lMED', 'Statut_PL'): 89.29525575772175,
 ('lIDE', 'Statut_PNL'): -16.37781477256297,
 ('lIDE', 'Statut_PL'): -32.143027743163785,
 ('lAID', 'Statut_PNL'): -70.22286505476518,
 ('lAID', 'Statut_PL'): -55.06680636513841,
 ('lADMIN', 'Statut_PNL'): -9.342415109437475,
 ('lADMIN', 'Statut_PL'): -14.98923007659625}

| Facteur   |    PL     |   PNL    |
|:----------|------------:|-------------:|
| ADMIN    |      -14.99 |        -9.34 |
| AID      |      -55.07 |       -70.22  |
| IDE      |      -32.14 |       -16.37 |
| MED      |      89.29 |         9.04 |

tableau regroupant les différences d'emploi de differences d'emploi de personnel à niveau de production égal selon la nature de l'etablissement (privé non lucratif ou privé lucratif) d'après les coefficients estimés. D'après les chiffres obtenus, les établissements privés sont plus économes en personnel administratif, en infirmiers et aides soignantes mais emploient plus de médecins à niveau de production égal.